src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,53 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendar-gap;
+  width: $month-width;
+  padding: 10px;
+
+  @for $days-count from 28 through 31 {
+    &--month-length--#{$days-count} {
+      & :nth-child(n + #{ $days-count + 1 }) {
+        display: none;
+      }
+    }
+  }
+
+  @each $day-name, $first-day, $weekend  in $start-days {
+    &--start-day--#{$day-name} {
+      & :first-child {
+        margin-left: $first-day + $first-day * $day-sides;
+      }
+    }
+  }
+
+  &__day {
+    height: $day-sides;
+    width: $day-sides;
+    background-color: $day-color;
+    border: 1px solid black;
+    transition: all 0.5s;
```

Comment:
Better to define which exact properties will be animated, because if you not define it, it will be applied to every property that can be animated